### PR TITLE
Fully implemented pretrained word vector model

### LINKED ISSUE #8: see how pretrained fastText model can improve performance
**Is your feature request related to a problem? Please describe.**
The current fastText model has to learn language from scratch, which is difficult/impossible with little training data

**Describe the solution you'd like**
I would like there to be a method to use a pretrained fastText model, downloading either the generic Wikipedia model or creating an unsupervised pretrained model trained on github issues.

**Additional context**
Pretrained English model available at  https://fasttext.cc/docs/en/english-vectors.html

### LINKED ISSUE #67: implement pretrained W2V into fasttext models

### LINKED ISSUE #83: Reduce fastText models size
**Is your feature request related to a problem? Please describe.**
fastText has proven itself to be quite useful in predicting github labels, but the models are way too big, especially for the pre-trained version. It is not convenient to save multiple models and load them in the app. Some care has already been taken to reduce model size, such as reducing vocabulary and vector size in the src/data/build_w2v_vocab notebook. More can still be done to significantly reduce the size.

**Describe the solution you'd like**

Two main ideas are to quantize the model, which fastText allows easily in their python package. Another option is to throw out even more vocabulary (e.g. if a word doesn't appear once in our target issues), and reduce vector size yet again. We could have a finetuned vector model that is useful for anyone, and smaller, specific ones per user.

### LINKED ISSUE #93: make k-fold loop outside training and use same dataset for both models
**Is your feature request related to a problem? Please describe.**
Currently to test between fasttext and SVM we create the subdatasets twice and perform k-fold cross validation on them. Since we take negative samples randomly, this brings up some variance in the process. If fasttext and SVM are using two different datasets that have variance in their predictability, we cannot honestly compare them. 

**Describe the solution you'd like**
I would like the functions to be altered so both models use the same dataset. Instead of using 5-fold cross validation we should randomize the negative sampling 5 times, then train the models on 80% and average the 20% of the datasets where we do our validation.

### PR DESCRIPTION
**Pull request by antter**
I reduced the model in a new notebook, added it to the pipeline, and changed the training loop

This is me implementing the experiments I was doing in the WIP PR into the main model. I will close that.

Fixes #93 
Fixes #8 
Fixes #83
Fixes #67

### CODE REVIEW

File: models/reduce_pretrained_ft.ipynb [hunk 1]
```diff
@@ -0,0 +1,401 @@
+{
```
Comment (aakankshaduggal) [inline models/reduce_pretrained_ft.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.        words = word_tokenize(preprocess(row.title + " SEP " + row.body if type(row.body) == str else row.title))
</pre><p>Could you please add an explanation here? Like a comment that explains why are we doing this.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/github-labeler/pull/95/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (aakankshaduggal) [inline models/reduce_pretrained_ft.ipynb:1]: <p>Why do we need to retain 70% of the total singular value sizes? Could you please add an explanation here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/github-labeler/pull/95/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

**Review by aakankshaduggal — COMMENTED** [review]
/lgtm
/approve
